### PR TITLE
[ADD] Add micro-benchmark for Cast

### PR DESCRIPTION
**Pull request by PeixuanZuo**
**Description**: Describe your changes.

Add micro-benchmark for Cast

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/python/tools/microbench/cast.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import argparse
+from dataclasses import dataclass
+from pydoc import ispackage
```
Comment (zhangyaobit) [inline onnxruntime/python/tools/microbench/cast.py:3]: Looks like ispackage is not used?

File: onnxruntime/python/tools/microbench/cast.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        outputs = {"Y": output_data}
+        return inputs, outputs
+
+    def create_cases(self):
+        test_data = self.create_test_input_shape()
+        for (x, y, m, n, is_fp16) in test_data:
+            model = "models/cast_fp16tofp32.onnx" if is_fp16 else "models/cast_fp32tofp16.onnx"
+            input_data_type = np.float16 if is_fp16 else np.float32
+            output_data_type = np.float32 if is_fp16 else np.float16
+            # bert-base
```
Comment (zhangyaobit) [inline onnxruntime/python/tools/microbench/cast.py:37]: delete?

File: onnxruntime/python/tools/microbench/cast.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            op_param = OpParam(x, y, m, n, input_data_type, output_data_type)
+            self.add_case(op_param, model)
+
+    def case_profile(cls, op_param, time):
+        profile = f"(x y m n input_data_type) = ({op_param.x} {op_param.y} {op_param.m} {op_param.n} {op_param.input_data_type}), {time:7.4f} ms"
+        return profile
+    
+    def create_test_input_shape(self):
+        test_input_shape = [
```
Comment (zhangyaobit) [inline onnxruntime/python/tools/microbench/cast.py:46]: Add a comment on the source (which model) of these shapes?

File: onnxruntime/python/tools/microbench/cast.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+class OpParam:
+    x : int
+    y : int
+    m : int
+    n : int
+    input_data_type : type
+    output_data_type : type
+    
+
+
```
Comment (zhangyaobit) [inline onnxruntime/python/tools/microbench/cast.py:18]: delete 1 blank line here?

**Review by zhangyaobit — APPROVED** [review]

**Comment by pengwa** [comment]
just FYI. maybe we should add the file prefix later as other files did in https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/. https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/onnxruntime_test.py

"#-------------------------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.
#--------------------------------------------------------------------------"

**Review by ytaous — APPROVED** [review]

**Comment by zhangyaobit** [comment]
> just FYI. maybe we should add the file prefix later as other files did in https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/. https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/onnxruntime_test.py
> 
> "------------------------------------------------------------------------- Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT License. --------------------------------------------------------------------------"

Yeah, that's good suggestion. Can you help add it Peixuan? Maybe in a separate PR.

**Comment by PeixuanZuo** [comment]
> 

OK, I merge this pr first and create a new pr to add the prefix

**Pull request merged**